model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -2200,14 +2173,107 @@ async def execute(self, user: User, request: ModelDownloadRequest) -> ModelDownl
         return ModelDownloadResponse(urls=urls)
 
 
-def infer_hardware_from_model_name(model_name: str) -> CreateDockerImageBatchJobResourceRequests:
+def _fill_hardware_info(
+    llm_artifact_gateway: LLMArtifactGateway, request: CreateLLMModelEndpointV1Request
+):
+    if (
+        request.gpus is None
+        or request.gpu_type is None
+        or request.cpus is None
+        or request.memory is None
+        or request.storage is None
+    ):
+        if not (
+            request.gpus is None
+            and request.gpu_type is None
+            and request.cpus is None
+            and request.memory is None
+            and request.storage is None
+        ):
+            raise ObjectHasInvalidValueException(
+                "All hardware spec fields (gpus, gpu_type, cpus, memory, storage) must be provided if any hardware spec field is missing."
```

Comment:
nit: "... if any hardware spec field is provided"?

Reply:
err, i think my previous comment is incorrect https://github.com/scaleapi/llm-engine/pull/515#discussion_r1597247261

Reply:
https://github.com/scaleapi/llm-engine/pull/515#discussion_r1597247261
i guess it works both ways. since here i'm only allowing two states: either all fields are provided, or none of them are provided